scrapy_po/middleware.py

```diff
@@ -21,45 +23,44 @@ class InjectPageObjectsMiddleware:
     """
     @inlineCallbacks
     def process_response(self, request: Request, response, spider):
-        # find out which arguments need to be created
+        # find out the dependencies
         callback = get_callback(request, spider)
-        kwargs_to_provide = andi.to_provide(callback,
-                                            can_provide=self.is_page_object)
+        providers = build_providers(response)
+        can_provide = can_provide_fn(providers)
+        plan = andi.plan(callback, can_provide,
+                         providers.__contains__)
```

Comment:
could you please use keyword arguments for arguments other than callback? I wonder if we should make them keyword-only in andi..

Reply:
Andi was changed. This was addressed here: https://github.com/scrapinghub/scrapy-po/pull/5/commits/166f48fe3b47fc7963629867bff9010b382c1bef